Sources/web3swift/Web3/Web3+Resolver.swift

```diff
@@ -0,0 +1,103 @@
+//
+//  Web3+Resolver.swift
+//  
+//
+//  Created by Jann Driessen on 01.11.22.
+//
+
+import Foundation
+import BigInt
+import Core
+
+public class PolicyResolver {
+    private let provider: Web3Provider
+
+    public init(provider: Web3Provider) {
+        self.provider = provider
+    }
+
+    public func resolveAll(for tx: inout CodableTransaction, with policies: Policies = .auto) async throws {
+        if tx.from != nil || tx.sender != nil {
+            // Nonce should be resolved first - as this might be needed for some
+            // tx's gas estimation
+            tx.nonce = try await resolveNonce(for: tx, with: policies.noncePolicy)
+        }
+
+        tx.gasLimit = try await resolveGasEstimate(for: tx, with: policies.gasLimitPolicy)
+
+        if case .eip1559 = tx.type {
+            tx.maxFeePerGas = await resolveGasBaseFee(for: policies.maxFeePerGasPolicy)
+            tx.maxPriorityFeePerGas = await resolveGasPriorityFee(for: policies.maxPriorityFeePerGasPolicy)
+        } else {
+            tx.gasPrice = await resolveGasPrice(for: policies.gasPricePolicy)
+        }
+    }
+
+    public func resolveGasBaseFee(for policy: FeePerGasPolicy) async -> BigUInt {
+        let oracle = Oracle(provider)
+        switch policy {
+        case .automatic:
+            return await oracle.baseFeePercentiles().max() ?? 0
+        case .manual(let value):
+            return value
+        }
+    }
+
+    public func resolveGasEstimate(for transaction: CodableTransaction, with policy: GasLimitPolicy) async throws -> BigUInt {
+        switch policy {
+        case .automatic, .withMargin:
```

Comment:
Margin is not applied. Is there something that needs to be done before it can be applied?

Reply:
Also that applies to other usages of `.withMargin`.